tests/variables.tf

```diff
@@ -0,0 +1,57 @@
+variable "build_id" {
+  default = ""
+  type    = "string"
+}
+
+variable "cloudwatch_agent_url_lx" {
+  default = "s3://amazoncloudwatch-agent/linux/amd64/latest/AmazonCloudWatchAgent.zip"
+  type    = "string"
+}
+
+variable "cloudwatch_agent_url_win" {
+  default = "s3://amazoncloudwatch-agent/windows/amd64/latest/AmazonCloudWatchAgent.zip"
+  type    = "string"
+}
+
+variable "instance_role" {
+  default = ""
+  type    = "string"
+}
+
+variable "keypair_name" {
+  default = ""
+  type    = "string"
+}
+
+variable "private_subnet" {
+  default = "10.0.0.0/8"
+  type    = "string"
+}
+
+variable "subnet_id" {
+  default = ""
+  type    = "string"
+}
+
+variable "security_group" {
+  default = ""
+  type    = "string"
+}
+
+variable "security_group_vpc" {
```

Comment:
I think this is unnecessary. If the user passes in a security group then it must already exist, and you don't need its vpc_id anywhere. If the user leaves `security_group` blank, then the module creates one.

To support the use case where a user passes in a `subnet_id`, and so the module-created `security_group` needs to be in _that_ VPC, you can use a conditional data resource for the `subnet_id` to lookup its `vpc_id`. See the example here:

* https://www.terraform.io/docs/providers/aws/d/subnet.html#example-usage

Reply:
To retrieve a value in a local from a conditional data resource, use the same trick we do for conditional outputs: `"${join("", data.<resource>.<name>.*.<attribute>)`